R/declare_estimator.R

```diff
@@ -261,44 +292,65 @@ validation_fn(model_handler) <-  function(ret, dots, label){
 estimator_handler <- tidy_estimator(model_handler)
 
 # called by model_handler, resets columns names !!!
-fit2tidy <- function(fit, coefficients = FALSE) {
-  summ <- coef(summary(fit))
-  summ <-
-    summ[, tolower(substr(colnames(summ), 1, 3)) %in% c("est", "std", "pr("), drop = FALSE]
-  ci <- suppressMessages(as.data.frame(confint(fit)))
-  return_data <-
-    data.frame(coefficient = rownames(summ),
-               summ,
-               ci,
-               stringsAsFactors = FALSE, row.names = NULL)
-  colnames(return_data) <- c("coefficient","est", "se", "p", "ci_lower", "ci_upper")
-
-
-  if (is.character(coefficients)) {
-    coefs_in_output <- coefficients %in% return_data$coefficient
+fit2tidy <- function(fit, term = FALSE) {
+  if (requireNamespace("broom", quietly = TRUE)) {
```

Comment:
should pull this chunk out into methods, similar to estimatr's fake broom https://github.com/DeclareDesign/estimatr/blob/master/R/S3_tidy.R.

Reply:
sorry, I don't understand -- this works on any fit object, so how can it be a method?

Reply:
we need to create s3 methods for each function we want to support as we do in estimatr, and rely on the already existing ones from estimatr, via UseMethod("tidy") instead of running directly in fit2tidy

Reply:
I think we can't do this, because the main thing we'd need to do is tidy.lm and tidy.glm - which would collide with those methods in broom. Do you agree @nfultz?

Reply:
by implementing DeclareDesign::tidy.default and masking estimatr::tidy.default, I think this would work. No need to reinvent tidy.lm or tidy.glm.

Reply:
ok thanks we'll try that